pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -2265,6 +2279,18 @@ private void deleteTableOnServer(String tableNameWithType) {
     }
   }
 
+  public void updateZkTimeInterval(TableConfig tableConfig, SegmentZKMetadata segmentZKMetadata,
+      Schema newSchema) {
+    String segmentName = segmentZKMetadata.getSegmentName();
+
+    String timeColumnName = tableConfig.getValidationConfig().getTimeColumnName();
+    if (StringUtils.isNotEmpty(timeColumnName)) {
+      DateTimeFieldSpec dateTimeFieldSpec = newSchema.getDateTimeSpec(timeColumnName);
```

Comment:
We should use `getSpecForTimeColumn(timeColumnName)` which can handle schema in old format